src/TypeScript/Analyzer/JavaScriptAnalyzer.cs

```diff
@@ -0,0 +1,130 @@
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using System;
+using System.Collections.Generic;
+using System.ComponentModel.Composition;
+using System.Diagnostics;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.VisualStudio.Threading;
+using SonarLint.VisualStudio.Core;
+using SonarLint.VisualStudio.Core.Analysis;
+using SonarLint.VisualStudio.Integration;
+using SonarLint.VisualStudio.TypeScript.EslintBridgeClient;
+using SonarLint.VisualStudio.TypeScript.EslintBridgeClient.Contract;
+
+namespace SonarLint.VisualStudio.TypeScript.Analyzer
+{
+    [Export(typeof(IAnalyzer))]
+    internal class JavaScriptAnalyzer : IAnalyzer
+    {
+        private readonly IEslintBridgeClient eslintBridgeClient;
+        private readonly IEslintBridgeIssueConverter issuesConverter;
+        private readonly ILogger logger;
+
+        [ImportingConstructor]
+        public JavaScriptAnalyzer(IEslintBridgeClient eslintBridgeClient, ILogger logger)
+            : this(eslintBridgeClient, new EslintBridgeIssueConverter(), logger)
+        {
+        }
+
+        internal JavaScriptAnalyzer(IEslintBridgeClient eslintBridgeClient, IEslintBridgeIssueConverter issuesConverter, ILogger logger)
+        {
+            this.eslintBridgeClient = eslintBridgeClient;
+            this.issuesConverter = issuesConverter;
+            this.logger = logger;
+        }
+
+        public bool IsAnalysisSupported(IEnumerable<AnalysisLanguage> languages)
+        {
+            return languages.Contains(AnalysisLanguage.Javascript);
+        }
+
+        public void ExecuteAnalysis(string path,
+            string charset,
+            IEnumerable<AnalysisLanguage> detectedLanguages,
+            IIssueConsumer consumer,
+            IAnalyzerOptions analyzerOptions,
+            CancellationToken cancellationToken)
+        {
+            Debug.Assert(IsAnalysisSupported(detectedLanguages));
+
+            ExecuteAnalysis(path, consumer, cancellationToken).Forget(); // fire and forget
+        }
+
+        internal async Task ExecuteAnalysis(string filePath, IIssueConsumer consumer, CancellationToken cancellationToken)
+        {
+            // Switch to a background thread
+            await TaskScheduler.Default;
+
+            try
+            {
+                var analysisResponse = await eslintBridgeClient.AnalyzeJs(filePath, cancellationToken);
```

Comment:
I'm not calling InitLinter yet in any place, I'm not sure where it should be called. Probably in the class that will be responsible for loading the user's settings (and call init-linter again if they change).

Reply:
We need to call init-linter:
* when eslint-bridge is started, and
* when the user settings change _if_ eslint-bridge is already running.

I think the classes loading the user's settings should be as thin as possible; their job should be to work out what the active settings are when asked. I don't think they should need to know about the eslint-bridge or track its status.

Maybe the `EslinttBridgeProcess`?

Reply:
I'll create a ticket for that.